From e79a96ff89b11fea03680633ab830894288ee9f0 Mon Sep 17 00:00:00 2001
From: Neal Frager <neal.frager@amd.com>
Date: Thu, 28 Nov 2024 07:50:28 +0000
Subject: [PATCH] arm64: versal: vek280 prebuilt pdi is incorrect

The prebuilt vpl_gen_fixed.pdi for the vek280 is incorrect and does not define
the DDR memory region for address 0x8 0000 0000 - 0x8 8000 0000.  For this
reason, the intree Linux dts memory node for the vek280 will not boot.

This patch works around the issue until the prebuilt pdi file is changed.

https://github.com/Xilinx/soc-prebuilt-firmware/blob/xilinx_v2024.2/vek280-versal/vpl_gen_fixed.pdi

Upstream: n/a

Signed-off-by: Neal Frager <neal.frager@amd.com>
---
 arch/arm64/boot/dts/xilinx/versal-vek280-revA.dts | 2 +-
 arch/arm64/boot/dts/xilinx/versal-vek280-revB.dts | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/arch/arm64/boot/dts/xilinx/versal-vek280-revA.dts b/arch/arm64/boot/dts/xilinx/versal-vek280-revA.dts
index ad5d590318e6..a306eacca5b3 100644
--- a/arch/arm64/boot/dts/xilinx/versal-vek280-revA.dts
+++ b/arch/arm64/boot/dts/xilinx/versal-vek280-revA.dts
@@ -19,7 +19,7 @@ / {
 
 	memory: memory@0 {
 		device_type = "memory";
-		reg = <0 0 0 0x80000000>, <0x8 0x0 0x1 0x80000000>; /* 2GB + 6GB */
+		reg = <0 0 0 0x80000000>, <0x8 0x80000000 0x1 0x80000000>; /* 2GB + 6GB */
 	};
 
 	chosen {
diff --git a/arch/arm64/boot/dts/xilinx/versal-vek280-revB.dts b/arch/arm64/boot/dts/xilinx/versal-vek280-revB.dts
index a5ee80a373e2..ba54a5ef0c73 100644
--- a/arch/arm64/boot/dts/xilinx/versal-vek280-revB.dts
+++ b/arch/arm64/boot/dts/xilinx/versal-vek280-revB.dts
@@ -19,7 +19,7 @@ / {
 
 	memory: memory@0 {
 		device_type = "memory";
-		reg = <0 0 0 0x80000000>, <0x8 0x00000000 0x1 0x80000000>; /* 2GB + 6GB */
+		reg = <0 0 0 0x80000000>, <0x8 0x80000000 0x1 0x80000000>; /* 2GB + 6GB */
 	};
 
 	chosen {
-- 
2.25.1

